paper/paper.md

```diff
@@ -0,0 +1,167 @@
+nbgrader has expanded to support a number of other features, including the
+ability to handle multiple courses on the same JupyterHub instance; the option
+to either include or hide autograder tests; customizable late penalties; and
+support for importing assignment files downloaded from a Learning Management
+System (LMS).
+
+# Statement of Need
+
+The use of computational methods has become increasingly widespread in fields
+outside of computer science [@wing2008computational]. As these disciplines begin
+to require computational tools, undergraduate curricula also begin to include
+topics in programming and computer science. However, because students are
+focused on the discipline that is the object of their study—and programming is
+likely a secondary interest—teaching students through traditional computer
+science offerings is not always effective [@Cortina2007; @Forte2005;
+@Guzdial2005]. While there are visual programming languages such as Raptor
+[@Carlisle2004] and Scratch [@Resnick2009] that are intended to be easy and
+enjoyable for those unfamiliar with programming to learn, they lack specialized
+tools such as numerical or data visualization libraries, which are required for
+effective work in domain sciences. A hybrid approach is to teach students
+computational concepts in an interactive environment where it is possible to
+quickly write, test, and tweak small units of code. Many such environments
+exist, including Mathematica [@mathematica], Maple [@maple], MATLAB [@matlab],
+Sage [@sage] and IPython [@PerezGranger2007].
+
+In recent years, the IPython project introduced the *Jupyter notebook*
+[@kluyver2016jupyter], an interface that is conducive to interactive and
+literate computing, where programmers can interleave prose with code and
+figures. The Jupyter notebook is ideal for educators because it allows them to
+create assignments which include instructions along with code or Markdown cells,
+in which students can provide solutions to exercises. Students can, for example,
+write code both to compute and visualize a particular result. Because the
+notebook is interactive, students can iterate on a coding problem without
+needing to switch back and forth between a command line and a text editor, and
+they can rapidly see results alongside the code which produced them.
+
+Instructors in many fields have begun using the Jupyter notebook as a teaching
+platform. The notebook has appeared in over 70 classes
+[@Castano_Jupyter_Map_Dataset] on subjects including geology, mathematics,
+mechanical engineering, data science, chemical engineering, and bioinformatics.
+Software Carpentry, an organization that aims to teach graduate students basic
+computational skills, has also adopted the notebook for some of its lessons
+[@Wilson2014].
+
+Despite its appearance in many classrooms, before to the existence of nbgrader,
+the notebook was rarely used on a large scale for *graded* assignments. Instead,
```

Comment:
(If notebooks were used with grading, it would have been largely graded old-style: print it out, and mark it by hand. I used this hybrid apprach for many years on non-IPython kernels).

Reply:
👍 for the suggestion, 😞 for having to print out notebooks